### ISSUE TITLE
Task 'testDebugUnitTest' not found

### ISSUE THREAD

**Issue by benjdero** [mxalbert1996/jacoco-android-gradle-plugin#3]
Hello, I have an Android library that use Jacoco for generating test reports. Since the original jacoco-android seems abandoned, I tried to switch to your fork so I could start using AGP 8. I'm stuck on an error that seems trivial but can't fix, and was hoping you could help me.

When I run a Gradle sync after switching to your fork I get the following error: 
```
A problem occurred configuring project ':core'.
> Could not create task ':core:jacocoTestDebugUnitTestReport'.
   > Task with name 'testDebugUnitTest' not found in project ':core'.
```

Which I don't understand because if I list the Gradle tasks of my project the task :core:testDebugUnitTest seems to exist.

**Comment by benjdero** [comment]
I made this dummy project which use a similar configuration to my real project: https://github.com/benjdero/JacocoBug

Last commit I tried to use AGP 8.1.0 and your jacoco-android plugin and got the error mentionned above.

The previous commit use AGP 7.4.2 and hiya's version of jacoco-android plugin and it's working, you can test it with `./gradlew -Pci --console=plain jacocoTestReport` for example.

**Comment by mxalbert1996** [comment]
Apparently it's because `org.barfuin.gradle.jacocolog` plugin is [eagerly realizing](https://gitlab.com/barfuin/gradle-jacoco-log/-/blob/master/src/main/java/org/barfuin/gradle/jacocolog/JacocoAggReportConfigAction.java#L223) jacoco report tasks, at which point the unit test tasks are not yet created.

The original plugin works because it uses the old variant API (which is supposed to be deleted in AGP 9.0) and that API runs after the unit test tasks are created.

I'm trying to find a workaround.

**Comment by mxalbert1996** [comment]
I can't find a workaround that doesn't break [task configuration avoidance](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html) so if you need `org.barfuin.gradle.jacocolog` I'm afraid you need to wait for [this issue](https://gitlab.com/barfuin/gradle-jacoco-log/-/issues/12).

**Comment by mxalbert1996** [comment]
Fixed in 0.2.1.

**Comment by benjdero** [comment]
Looks good to me. Good job!
